DATA CLEANING IN PYTHON

In [1]:
import pandas as pd

In [14]:
df = pd.read_excel (r"C:\Users\Julius\Desktop\Portfolio project\Python\Customer Call List.xlsx")
df


CustomerID First_Name    Last_Name  Phone_Number  \
0         1001      Frodo      Baggins  123-545-5421   
1         1002       Abed        Nadir  123/643/9775   
2         1003     Walter       /White    7066950392   
3         1004     Dwight      Schrute  123-543-2345   
4         1005        Jon         Snow  876|678|3469   
5         1006        Ron      Swanson  304-762-2467   
6         1007       Jeff       Winger           NaN   
7         1008   Sherlock       Holmes  876|678|3469   
8         1009    Gandalf          NaN           N/a   
9         1010      Peter       Parker  123-545-5421   
10        1011    Samwise       Gamgee           NaN   
11        1012      Harry    ...Potter    7066950392   
12        1013        Don       Draper  123-543-2345   
13        1014     Leslie        Knope  876|678|3469   
14        1015       Toby  Flenderson_  304-762-2467   
15        1016        Ron      Weasley  123-545-5421   
16        1017   Michael         Scott  123/643/9775   
17        1018      Clark         Kent    7066950392   
18        1019      Creed       Braton           N/a   
19        1020     Anakin    Skywalker  876|678|3469   
20        1020     Anakin    Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   
20            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True  
20               True

In [18]:
df=df.drop_duplicates ()
df

CustomerID First_Name    Last_Name  Phone_Number  \
0         1001      Frodo      Baggins  123-545-5421   
1         1002       Abed        Nadir  123/643/9775   
2         1003     Walter       /White    7066950392   
3         1004     Dwight      Schrute  123-543-2345   
4         1005        Jon         Snow  876|678|3469   
5         1006        Ron      Swanson  304-762-2467   
6         1007       Jeff       Winger           NaN   
7         1008   Sherlock       Holmes  876|678|3469   
8         1009    Gandalf          NaN           N/a   
9         1010      Peter       Parker  123-545-5421   
10        1011    Samwise       Gamgee           NaN   
11        1012      Harry    ...Potter    7066950392   
12        1013        Don       Draper  123-543-2345   
13        1014     Leslie        Knope  876|678|3469   
14        1015       Toby  Flenderson_  304-762-2467   
15        1016        Ron      Weasley  123-545-5421   
16        1017   Michael         Scott  123/643/9775   
17        1018      Clark         Kent    7066950392   
18        1019      Creed       Braton           N/a   
19        1020     Anakin    Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True

In [ ]:
df = df.drop(columns = "Not_Useful_Column")
df

In [38]:
df


CustomerID First_Name    Last_Name  Phone_Number  \
0         1001      Frodo      Baggins  123-545-5421   
1         1002       Abed        Nadir  123/643/9775   
2         1003     Walter       /White    7066950392   
3         1004     Dwight      Schrute  123-543-2345   
4         1005        Jon         Snow  876|678|3469   
5         1006        Ron      Swanson  304-762-2467   
6         1007       Jeff       Winger           NaN   
7         1008   Sherlock       Holmes  876|678|3469   
8         1009    Gandalf          NaN           N/a   
9         1010      Peter       Parker  123-545-5421   
10        1011    Samwise       Gamgee           NaN   
11        1012      Harry    ...Potter    7066950392   
12        1013        Don       Draper  123-543-2345   
13        1014     Leslie        Knope  876|678|3469   
14        1015       Toby  Flenderson_  304-762-2467   
15        1016        Ron      Weasley  123-545-5421   
16        1017   Michael         Scott  123/643/9775   
17        1018      Clark         Kent    7066950392   
18        1019      Creed       Braton           N/a   
19        1020     Anakin    Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  
0                   123 Shire Lane, Shire             Yes             No  
1                     93 West Main Street              No            Yes  
2                      298 Drugs Driveway               N            NaN  
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y  
4                        123 Dragons Road               Y             No  
5                        768 City Parkway             Yes            Yes  
6                       1209 South Street              No             No  
7                           98 Clue Drive               N             No  
8                        123 Middle Earth             Yes            NaN  
9              25th Main Street, New York             Yes             No  
10                  612 Shire Lane, Shire             Yes             No  
11                   2394 Hogwarts Avenue               Y            NaN  
12                       2039 Main Street             Yes              N  
13                       343 City Parkway             Yes             No  
14                          214 HR Avenue               N             No  
15                   2395 Hogwarts Avenue              No              N  
16         121 Paper Avenue, Pennsylvania             Yes             No  
17                        3498 Super Lane               Y            NaN  
18                                    N/a             N/a            Yes  
19            910 Tatooine Road, Tatooine             Yes              N

YOU CAN ALSO STRIP AS BELOW
df ["Last_Name"] = df ["Last_Name"].str.strip("123./_") and then call df

In [58]:
df ["Last_Name"] = df ["Last_Name"].str.strip("...")
df ["Last_Name"] = df ["Last_Name"].str.strip("/")
df ["Last_Name"] = df ["Last_Name"].str.strip("_")
df ["Last_Name"] = df ["Last_Name"].str.strip("...")

df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed       Nadir  123/643/9775   
2         1003     Walter       White    7066950392   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876|678|3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger           NaN   
7         1008   Sherlock      Holmes  876|678|3469   
8         1009    Gandalf         NaN           N/a   
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee           NaN   
11        1012      Harry      Potter    7066950392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876|678|3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123/643/9775   
17        1018      Clark        Kent    7066950392   
18        1019      Creed      Braton           N/a   
19        1020     Anakin   Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  
0                   123 Shire Lane, Shire             Yes             No  
1                     93 West Main Street              No            Yes  
2                      298 Drugs Driveway               N            NaN  
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y  
4                        123 Dragons Road               Y             No  
5                        768 City Parkway             Yes            Yes  
6                       1209 South Street              No             No  
7                           98 Clue Drive               N             No  
8                        123 Middle Earth             Yes            NaN  
9              25th Main Street, New York             Yes             No  
10                  612 Shire Lane, Shire             Yes             No  
11                   2394 Hogwarts Avenue               Y            NaN  
12                       2039 Main Street             Yes              N  
13                       343 City Parkway             Yes             No  
14                          214 HR Avenue               N             No  
15                   2395 Hogwarts Avenue              No              N  
16         121 Paper Avenue, Pennsylvania             Yes             No  
17                        3498 Super Lane               Y            NaN  
18                                    N/a             N/a            Yes  
19            910 Tatooine Road, Tatooine             Yes              N

In [146]:
#df ["Phone_Number"].str.replace ("[^a-zA-Z0-9]", ".")
#df ["phone_Number"].apply (lambda x: x[0:3] + "-" + x[3:6] + "-" + x[6:10])
#df ["Phone_Number"]. apply (lambda x: str (x))
#df ["phone_Number"].apply(lambda x: x[0:3] + '-' + x[3:6] + '-' + x[6:10])
#df

df ["Phone_Number"] = df ["Phone_Number"].str.replace ("N/a", " ")
df ["Phone_Number"] = df ["Phone_Number"].str.replace ("NaN", " ")
df ["Phone_Number"]. apply (lambda x: str (x))
df ["Phone_Number"] = df ["Phone_Number"].str.replace ("nan--", " ")



In [156]:
df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed       Nadir  123/643/9775   
2         1003     Walter       White           NaN   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876|678|3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger           NaN   
7         1008   Sherlock      Holmes  876|678|3469   
8         1009    Gandalf         NaN                 
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee           NaN   
11        1012      Harry      Potter           NaN   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876|678|3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123/643/9775   
17        1018      Clark        Kent           NaN   
18        1019      Creed      Braton                 
19        1020     Anakin   Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  
0                   123 Shire Lane, Shire             Yes             No  
1                     93 West Main Street              No            Yes  
2                      298 Drugs Driveway               N            NaN  
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y  
4                        123 Dragons Road               Y             No  
5                        768 City Parkway             Yes            Yes  
6                       1209 South Street              No             No  
7                           98 Clue Drive               N             No  
8                        123 Middle Earth             Yes            NaN  
9              25th Main Street, New York             Yes             No  
10                  612 Shire Lane, Shire             Yes             No  
11                   2394 Hogwarts Avenue               Y            NaN  
12                       2039 Main Street             Yes              N  
13                       343 City Parkway             Yes             No  
14                          214 HR Avenue               N             No  
15                   2395 Hogwarts Avenue              No              N  
16         121 Paper Avenue, Pennsylvania             Yes             No  
17                        3498 Super Lane               Y            NaN  
18                                    N/a             N/a            Yes  
19            910 Tatooine Road, Tatooine             Yes              N

In [170]:
df[["Street_adress", "State", "Zip_code"]]=df["Address"].str.split(',' , expand = True)
df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed       Nadir  123/643/9775   
2         1003     Walter       White           NaN   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876|678|3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger           NaN   
7         1008   Sherlock      Holmes  876|678|3469   
8         1009    Gandalf         NaN                 
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee           NaN   
11        1012      Harry      Potter           NaN   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876|678|3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123/643/9775   
17        1018      Clark        Kent           NaN   
18        1019      Creed      Braton                 
19        1020     Anakin   Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   

           Street_adress          State Zip_code  
0         123 Shire Lane          Shire     None  
1    93 West Main Street           None     None  
2     298 Drugs Driveway           None     None  
3       980 Paper Avenue   Pennsylvania    18503  
4       123 Dragons Road           None     None  
5       768 City Parkway           None     None  
6      1209 South Street           None     None  
7          98 Clue Drive           None     None  
8       123 Middle Earth           None     None  
9       25th Main Street       New York     None  
10        612 Shire Lane          Shire     None  
11  2394 Hogwarts Avenue           None     None  
12      2039 Main Street           None     None  
13      343 City Parkway           None     None  
14         214 HR Avenue           None     None  
15  2395 Hogwarts Avenue           None     None  
16      121 Paper Avenue   Pennsylvania     None  
17       3498 Super Lane           None     None  
18                   N/a           None     None  
19     910 Tatooine Road       Tatooine     None

In [ ]:
df = df.drop(columns=['Address'])

In [178]:
df

CustomerID First_Name   Last_Name  Phone_Number Paying Customer  \
0         1001      Frodo     Baggins  123-545-5421             Yes   
1         1002       Abed       Nadir  123/643/9775              No   
2         1003     Walter       White           NaN               N   
3         1004     Dwight     Schrute  123-543-2345             Yes   
4         1005        Jon        Snow  876|678|3469               Y   
5         1006        Ron     Swanson  304-762-2467             Yes   
6         1007       Jeff      Winger           NaN              No   
7         1008   Sherlock      Holmes  876|678|3469               N   
8         1009    Gandalf         NaN                           Yes   
9         1010      Peter      Parker  123-545-5421             Yes   
10        1011    Samwise      Gamgee           NaN             Yes   
11        1012      Harry      Potter           NaN               Y   
12        1013        Don      Draper  123-543-2345             Yes   
13        1014     Leslie       Knope  876|678|3469             Yes   
14        1015       Toby  Flenderson  304-762-2467               N   
15        1016        Ron     Weasley  123-545-5421              No   
16        1017   Michael        Scott  123/643/9775             Yes   
17        1018      Clark        Kent           NaN               Y   
18        1019      Creed      Braton                           N/a   
19        1020     Anakin   Skywalker  876|678|3469             Yes   

   Do_Not_Contact         Street_adress          State Zip_code  
0              No        123 Shire Lane          Shire     None  
1             Yes   93 West Main Street           None     None  
2             NaN    298 Drugs Driveway           None     None  
3               Y      980 Paper Avenue   Pennsylvania    18503  
4              No      123 Dragons Road           None     None  
5             Yes      768 City Parkway           None     None  
6              No     1209 South Street           None     None  
7              No         98 Clue Drive           None     None  
8             NaN      123 Middle Earth           None     None  
9              No      25th Main Street       New York     None  
10             No        612 Shire Lane          Shire     None  
11            NaN  2394 Hogwarts Avenue           None     None  
12              N      2039 Main Street           None     None  
13             No      343 City Parkway           None     None  
14             No         214 HR Avenue           None     None  
15              N  2395 Hogwarts Avenue           None     None  
16             No      121 Paper Avenue   Pennsylvania     None  
17            NaN       3498 Super Lane           None     None  
18            Yes                   N/a           None     None  
19              N     910 Tatooine Road       Tatooine     None

In [200]:
df['Paying Customer']=df['Paying Customer'].replace('Yes', 'Y')
df['Paying Customer']=df['Paying Customer'].replace('No', 'N')
df['Do_Not_Contact']=df['Do_Not_Contact'].replace('Yes', 'Y')
df['Do_Not_Contact']=df['Do_Not_Contact'].replace('No', 'N')
df

CustomerID First_Name   Last_Name  Phone_Number Paying Customer  \
0         1001      Frodo     Baggins  123-545-5421               Y   
1         1002       Abed       Nadir  123/643/9775               N   
2         1003     Walter       White           NaN               N   
3         1004     Dwight     Schrute  123-543-2345               Y   
4         1005        Jon        Snow  876|678|3469               Y   
5         1006        Ron     Swanson  304-762-2467               Y   
6         1007       Jeff      Winger           NaN               N   
7         1008   Sherlock      Holmes  876|678|3469               N   
8         1009    Gandalf         NaN                             Y   
9         1010      Peter      Parker  123-545-5421               Y   
10        1011    Samwise      Gamgee           NaN               Y   
11        1012      Harry      Potter           NaN               Y   
12        1013        Don      Draper  123-543-2345               Y   
13        1014     Leslie       Knope  876|678|3469               Y   
14        1015       Toby  Flenderson  304-762-2467               N   
15        1016        Ron     Weasley  123-545-5421               N   
16        1017   Michael        Scott  123/643/9775               Y   
17        1018      Clark        Kent           NaN               Y   
18        1019      Creed      Braton                           N/a   
19        1020     Anakin   Skywalker  876|678|3469               Y   

   Do_Not_Contact         Street_adress          State Zip_code  
0               N        123 Shire Lane          Shire     None  
1               Y   93 West Main Street           None     None  
2             NaN    298 Drugs Driveway           None     None  
3               Y      980 Paper Avenue   Pennsylvania    18503  
4               N      123 Dragons Road           None     None  
5               Y      768 City Parkway           None     None  
6               N     1209 South Street           None     None  
7               N         98 Clue Drive           None     None  
8             NaN      123 Middle Earth           None     None  
9               N      25th Main Street       New York     None  
10              N        612 Shire Lane          Shire     None  
11            NaN  2394 Hogwarts Avenue           None     None  
12              N      2039 Main Street           None     None  
13              N      343 City Parkway           None     None  
14              N         214 HR Avenue           None     None  
15              N  2395 Hogwarts Avenue           None     None  
16              N      121 Paper Avenue   Pennsylvania     None  
17            NaN       3498 Super Lane           None     None  
18              Y                   N/a           None     None  
19              N     910 Tatooine Road       Tatooine     None

In [218]:
df=df.fillna(' ')
df


CustomerID First_Name   Last_Name  Phone_Number Paying Customer  \
0         1001      Frodo     Baggins  123-545-5421               Y   
1         1002       Abed       Nadir  123/643/9775               N   
2         1003     Walter       White                             N   
3         1004     Dwight     Schrute  123-543-2345               Y   
4         1005        Jon        Snow  876|678|3469               Y   
5         1006        Ron     Swanson  304-762-2467               Y   
6         1007       Jeff      Winger                             N   
7         1008   Sherlock      Holmes  876|678|3469               N   
8         1009    Gandalf                                         Y   
9         1010      Peter      Parker  123-545-5421               Y   
10        1011    Samwise      Gamgee                             Y   
11        1012      Harry      Potter                             Y   
12        1013        Don      Draper  123-543-2345               Y   
13        1014     Leslie       Knope  876|678|3469               Y   
14        1015       Toby  Flenderson  304-762-2467               N   
15        1016        Ron     Weasley  123-545-5421               N   
16        1017   Michael        Scott  123/643/9775               Y   
17        1018      Clark        Kent                             Y   
18        1019      Creed      Braton                           N/a   
19        1020     Anakin   Skywalker  876|678|3469               Y   

   Do_Not_Contact         Street_adress          State Zip_code  
0               N        123 Shire Lane          Shire           
1               Y   93 West Main Street                          
2                    298 Drugs Driveway                          
3               Y      980 Paper Avenue   Pennsylvania    18503  
4               N      123 Dragons Road                          
5               Y      768 City Parkway                          
6               N     1209 South Street                          
7               N         98 Clue Drive                          
8                      123 Middle Earth                          
9               N      25th Main Street       New York           
10              N        612 Shire Lane          Shire           
11                 2394 Hogwarts Avenue                          
12              N      2039 Main Street                          
13              N      343 City Parkway                          
14              N         214 HR Avenue                          
15              N  2395 Hogwarts Avenue                          
16              N      121 Paper Avenue   Pennsylvania           
17                      3498 Super Lane                          
18              Y                   N/a                          
19              N     910 Tatooine Road       Tatooine

In [226]:
df=df.replace ('N/a', '')
df

CustomerID First_Name   Last_Name  Phone_Number Paying Customer  \
0         1001      Frodo     Baggins  123-545-5421               Y   
1         1002       Abed       Nadir  123/643/9775               N   
2         1003     Walter       White                             N   
3         1004     Dwight     Schrute  123-543-2345               Y   
4         1005        Jon        Snow  876|678|3469               Y   
5         1006        Ron     Swanson  304-762-2467               Y   
6         1007       Jeff      Winger                             N   
7         1008   Sherlock      Holmes  876|678|3469               N   
8         1009    Gandalf                                         Y   
9         1010      Peter      Parker  123-545-5421               Y   
10        1011    Samwise      Gamgee                             Y   
11        1012      Harry      Potter                             Y   
12        1013        Don      Draper  123-543-2345               Y   
13        1014     Leslie       Knope  876|678|3469               Y   
14        1015       Toby  Flenderson  304-762-2467               N   
15        1016        Ron     Weasley  123-545-5421               N   
16        1017   Michael        Scott  123/643/9775               Y   
17        1018      Clark        Kent                             Y   
18        1019      Creed      Braton                                 
19        1020     Anakin   Skywalker  876|678|3469               Y   

   Do_Not_Contact         Street_adress          State Zip_code  
0               N        123 Shire Lane          Shire           
1               Y   93 West Main Street                          
2                    298 Drugs Driveway                          
3               Y      980 Paper Avenue   Pennsylvania    18503  
4               N      123 Dragons Road                          
5               Y      768 City Parkway                          
6               N     1209 South Street                          
7               N         98 Clue Drive                          
8                      123 Middle Earth                          
9               N      25th Main Street       New York           
10              N        612 Shire Lane          Shire           
11                 2394 Hogwarts Avenue                          
12              N      2039 Main Street                          
13              N      343 City Parkway                          
14              N         214 HR Avenue                          
15              N  2395 Hogwarts Avenue                          
16              N      121 Paper Avenue   Pennsylvania           
17                      3498 Super Lane                          
18              Y                                                
19              N     910 Tatooine Road       Tatooine

# Filter and display only IDs where 'Do_Not_Contact' is 'N' (replace 'ID_column' with actual column name)

In [228]:

call_ids = df[df['Do_Not_Contact'] == 'N']['CustomerID']
print(call_ids)

0     1001
4     1005
6     1007
7     1008
9     1010
10    1011
12    1013
13    1014
14    1015
15    1016
16    1017
19    1020
Name: CustomerID, dtype: int64


# Filter and display only Customers where 'Do_Not_Contact' is 'N' (replace 'ID_column' with actual column name)

In [236]:
call_customers = df[df['Do_Not_Contact'] == 'N'][['CustomerID', 'First_Name', 'Last_Name', 'Phone_Number']]
print(call_customers)

    CustomerID First_Name   Last_Name  Phone_Number
0         1001      Frodo     Baggins  123-545-5421
4         1005        Jon        Snow  876|678|3469
6         1007       Jeff      Winger              
7         1008   Sherlock      Holmes  876|678|3469
9         1010      Peter      Parker  123-545-5421
10        1011    Samwise      Gamgee              
12        1013        Don      Draper  123-543-2345
13        1014     Leslie       Knope  876|678|3469
14        1015       Toby  Flenderson  304-762-2467
15        1016        Ron     Weasley  123-545-5421
16        1017   Michael        Scott  123/643/9775
19        1020     Anakin   Skywalker  876|678|3469


In [240]:
df=call_customers

In [242]:
df

CustomerID First_Name   Last_Name  Phone_Number
0         1001      Frodo     Baggins  123-545-5421
4         1005        Jon        Snow  876|678|3469
6         1007       Jeff      Winger              
7         1008   Sherlock      Holmes  876|678|3469
9         1010      Peter      Parker  123-545-5421
10        1011    Samwise      Gamgee              
12        1013        Don      Draper  123-543-2345
13        1014     Leslie       Knope  876|678|3469
14        1015       Toby  Flenderson  304-762-2467
15        1016        Ron     Weasley  123-545-5421
16        1017   Michael        Scott  123/643/9775
19        1020     Anakin   Skywalker  876|678|3469

In [254]:
file_path = "output.xlsx"

In [256]:
df.to_excel(file_path, index=False)

In [258]:
print(f"DataFrame has been exported to {file_path}")

DataFrame has been exported to output.xlsx


In [262]:
export = (r"C:\Users\Julius\Desktop\Portfolio project\Python\Clean_CustomerCall List.xlsx")

In [264]:
df.to_excel(export, index=False)

In [266]:
print(f"DataFrame has been exported to {export}")

DataFrame has been exported to C:\Users\Julius\Desktop\Portfolio project\Python\Clean_CustomerCall List.xlsx
